onnxruntime/python/onnxruntime_pybind_mlvalue.h

```diff
@@ -21,14 +21,21 @@ namespace python {
 
 namespace py = pybind11;
 
+bool PyObjectCheck_Array(PyObject* o);
+
 int OnnxRuntimeTensorToNumpyType(const DataTypeImpl* tensor_type);
 
 MLDataType NumpyTypeToOnnxRuntimeType(int numpy_type);
 
-void CreateGenericMLValue(const onnxruntime::InputDefList* input_def_list, const AllocatorPtr& alloc, const std::string& name_input,
-                          py::object& value, OrtValue* p_mlvalue);
+using MemCpyFunc = void (*)(void*, const void*, size_t);
+void CpuToCpuMemCpy(void*, const void*, size_t);
+void CreateGenericMLValue(const onnxruntime::InputDefList* input_def_list, const AllocatorPtr& alloc,
+                          const std::string& name_input, py::object& value, OrtValue* p_mlvalue,
+                          bool parse_numpy_only = false, bool use_numpy_data_memory = true, MemCpyFunc mem_cpy_to_device = CpuToCpuMemCpy);
```

Comment:
Indirectly yes. The goal of CreateGenericMLValue() is to parse the provided python object (could be a numpy array, could be a dictionary, could be a list of numpy arrays) and create an OrtValue from that. So in the context of this function, I felt a flag named 'parse_numpy_only' seemed more contextually correct instead of something like 'parse_tensors'.

---
In reply to: [493250220](https://github.com/microsoft/onnxruntime/pull/5248#discussion_r493250220) [](ancestors = 493250220)